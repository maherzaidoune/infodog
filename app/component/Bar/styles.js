import {StyleSheet, Dimensions} from 'react-native';

const WIDTH = Dimensions.get('window').width;
const HEIGHT = Dimensions.get('window').height;
export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingStart: 4,
    paddingEnd: 4,
  },
  chartContainer: {
    height: WIDTH / 2,
  },
  scrollContainer: {
    // padding: 20,
  },
  barStyle: {
    height: WIDTH / 2,
  },
});
