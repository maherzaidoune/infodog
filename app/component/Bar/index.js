import React, {useRef, useEffect} from 'react';
import {ScrollView, View} from 'react-native';
import {BarChart, Grid, YAxis} from 'react-native-svg-charts';
import {colors} from '../../constants/colors';
import {styles} from './styles';

const Bar = ({data = [], formatLabel = label => label}) => {
  const scrollview = useRef();
  const isScrolling = useRef(false);

  const Yaxis = data.map(i => Math.floor(i));

  const onUpdate = () => {
    //block autoscrolling when user scrolling so user can look into older metrics
    if (!isScrolling.current) {
      scrollview.current.scrollToEnd({animated: true});
    }
  };

  const onScrollBegin = () => {
    isScrolling.current = true;
  };

  const onScrollEnd = () => {
    setTimeout(() => {
      isScrolling.current = false;
    }, 1000);
  };

  useEffect(() => {
    onUpdate();
  }, [data]);

  return (
    <View style={styles.container}>
      <YAxis
        data={Yaxis}
        contentInset={{top: 0, bottom: 10, left: 10, right: 0}}
        svg={{
          fill: 'grey',
          fontSize: 10,
        }}
        formatLabel={formatLabel}
      />
      <ScrollView
        ref={scrollview}
        horizontal={true}
        showsHorizontalScrollIndicator={false}
        removeClippedSubviews
        onScrollBeginDrag={onScrollBegin}
        onScrollEndDrag={onScrollEnd}
        contentContainerStyle={styles.scrollContainer}>
        <View style={[styles.chartContainer, {width: data.length * 30}]}>
          <BarChart
            style={[styles.barStyle, {width: data.length * 30}]}
            data={data}
            svg={{fill: colors.main}}
            contentInset={{top: 0, left: 4, right: 0}}>
            <Grid />
          </BarChart>
        </View>
      </ScrollView>
    </View>
  );
};

export default React.memo(Bar);
