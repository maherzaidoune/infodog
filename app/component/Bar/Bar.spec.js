import React from 'react';
import Bar from '.';

import renderer from 'react-test-renderer';

test('Bar renders correctly', () => {
  const tree = renderer.create(<Bar data={[]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
