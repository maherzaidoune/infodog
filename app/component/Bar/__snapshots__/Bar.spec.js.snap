// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingEnd": 4,
      "paddingStart": 4,
    }
  }
>
  <View />
  <RCTScrollView
    contentContainerStyle={Object {}}
    horizontal={true}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    showsHorizontalScrollIndicator={true}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "height": 375,
            },
            Object {
              "width": 0,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 375,
              },
              Object {
                "width": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
